Classes/Service/AuthenticationService.php

```diff
@@ -369,17 +370,16 @@ protected function convertResourceOwner(array $info): bool|array
             $row ?: [],
             [
                 'tx_oidc' => $info['sub'],
-                'password' => $this->generatePassword(),
                 'deleted' => 0,
                 'disable' => 0,
             ]
         );
 
-        // preserve username and password for existing users
+        // preserve password for existing users, an existing password should not be replaced on every login
         if ($row) {
-            unset($data['username']);
-            unset($data['email']);
             unset($data['password']);
+        } else {
+            $data['password'] = $this->generatePassword();
```

Comment:
We could do this here: https://github.com/xperseguers/t3ext-oidc/blob/master/Classes/Service/AuthenticationService.php#L501 where the other initial values (pid, crdate) are set.

Reply:
I was undecided whether to keep the password-handling in one place or the user-creation. :-) 
But I'll move it down.

Reply:
I am not sure if we need the `unset` at all for existing users. The password is not part of the default mapping. So if somebody wants to intentionally set it via TypoScript, why not let them?

Reply:
it will be set, because it is prefilled from $row. This line is a pure safety net to avoid messing with the password field by any means.
Setting it via TS is not possible, as you would need to run it through the hashing stuff first

Reply:
> This line is a pure safety net to avoid messing with the password field by any means.

OK, maybe adjust the comment accordingly. "This line disallows the integrator to mess with the password" 

> Setting it via TS is not possible, as you would need to run it through the hashing stuff first

It would be via userFunc or with fe_users without hashing. But I agree that this would be pretty weird.

Reply:
Will enhance the comment.

> It would be via userFunc or with fe_users without hashing. But I agree that this would be pretty weird.

`userFunc`: indeed 😇

"without hashing": I think this is not even possible anymore without implementing your own authentication service. 😁